amx/server/syscalls.lua

```diff
@@ -2763,15 +2763,37 @@ function SetPlayerChatBubble(amx, player, text, color, dist, exptime)
 end
 
 
--- In MTA we can't set the type of the request. 
--- When we add 'data' it becomes POST request.
--- TODO: arguments "index" and "type" here only for compatibility.
+-- Now we have all of RESTful types of requests. Our function is better!
+-- The SAMP documentation said about 'url' - "The URL you want to request. (Without 'http://')"
+-- But I didn't set it up. Because WE SUPPORT HTTPS!
+-- TODO: An "index" argument only for compatibility.
 function HTTP(amx, index, type, url, data, callback)
-	if type == 3 then 
-		notImplemented('HTTP', 'A HEAD type have not support right now')
-	end
-	local request = fetchRemote ("http://".. url, 
-	function(responseData, error)
+
+	local typesToText = {
+		'GET',
+		'POST',
+		'HEAD',
+		'PUT',
+		'PATCH',
+		'DELETE',
+		'COPY',
+		'OPTIONS',
+		'LINK',
+		'UNLINK',
+		'PURGE',
+		'LOCK',
+		'UNLOCK',
+		'PROPFIND',
+		'VIEW'
+	}
+	local sendOptions = {
+		queueName = "A SAMP request queue",

```

Comment:
Could use `"amx." .. getResourceName(amx.res) .. "." .. amx.name`?

Then each amx file in each resource gets their own queue.

But how does SAMP do it? Does it queue per plugin or use a global queue?

Reply:
> But how does SAMP do it?

I guess we won't know about it ever.

I try to use first variant. If it won't work I'll use `queueName = "amx"`

Reply:
> I guess we won't know about it ever.

We could figure it out by sending requests to a (local / our own / a custom) webserver and see how the queue behaves in SA-MP.

But yeah, that will take some time, so I think it's fine if we investigate that later and not bother about this right now.